autoPVS1_from_VEP_vcf.py

```diff
@@ -1,48 +1,110 @@
 
 
-lof_type = ['frameshift', 'nonsense', 'splice-5', 'splice-3', 'init-loss']
-vep_lof_list = ['frameshift', 'stop_gained', 'splice_donor', 'splice_acceptor', 'start_lost']
-VAR = namedtuple('VAR', ('varid', 'gene', 'trans', 'canonical', 'pick', 'record'))
-
-
-class AutoPVS1:
-    """Run AutoPVS1"""
-    @staticmethod
-    def get_vcfrecord(vcf):
-        if all(hasattr(vcf, attr) for attr in ['chrom', 'pos', 'ref', 'alt']):
-            return vcf
-        elif re.match(r'^((?:chr)?[\dXYMT]{1,2})-(\d+)-([ATCG]+)-([ATCG]+)$', vcf, re.I):
-            v = vcf.split("-")
-            v[0] = re.sub('chr', '', v[0], flags=re.I)
-            return VCFRecord(v[0], int(v[1]), v[2], v[3])
-        else:
-            raise TypeError("Wrong VCF Record")
-
-
-    def run_pvs1(self):
-        pvs1 = PVS1(self.vcfrecord, self.consequence, self.hgvs_c, 
-                    self.hgvs_p, self.transcript, self.genome_version)
-        return pvs1
+def get_candidates(record_csq, csq_fields):
+    """
+    1. See if the tx candidate is canonical
+    2. If canonical, check to see if it exists in the tool transcript reference
+    3. Record the PICK value for rule 5 from pick_transcript
+    """
+    canon_idx = csq_fields.index("CANONICAL")
+    p_idx = csq_fields.index("PICK")
+    
+    pick_value = "1"
+    canon_value = "YES"
+    r_idx = csq_fields.index('RefSeq')
+    picked_csqs = []
+    tx_candidates = []
+
+    for csq in record_csq:
+        info_list = csq.split("|")
+        refseq_id = info_list[r_idx]
+        if info_list[p_idx] == pick_value:
+            pick = (get_transcript(refseq_id, transcripts_hg38), info_list)
+        if info_list[canon_idx] == canon_value:
+            if refseq_id != "":
+                # pdb.set_trace()
+                check = get_transcript(refseq_id, transcripts_hg38)
+                if check is not None:
+                    picked_csqs.append(info_list)
+                    tx_candidates.append(check)
+    return picked_csqs, tx_candidates, pick
+
+
+def pick_transcript(record, csq_fields, summary, csq_severity_order):
+    """
+    Function to process a record and choose a representative transcript
+    Does so by doing the following:
+    1. See if the tx candidate is canonical
+    2. If canonical, check to see if it exists in the tool transcript reference
+    3. If multiple hits exists, use rank list to chose the worst
+    4. If still multiple (meaning ranks are tied), use longest transcript length as defined by the reference
+    5. If no hits from step 1 and 2, just revert to using PICK
+    """
+    picked_csqs, tx_candidates, pick = get_candidates(record.info['CSQ'], csq_fields)
+    csq_idx = csq_fields.index("Consequence")
+
+    # tie break with rank - if multiple share rank, go with PICK, if not go with tx length
+    if len(picked_csqs) > 1:
+        best_rank = 1000
+        champ = 0
+        summary['rank'] += 1
+        score_dict = {}
+        for c in range(len(picked_csqs)):
```

Comment:
I think you're trying to `enumerate` here:
```
>>> l = ["a","b","c"]
>>> for x,y in enumerate(l):
...     print(x,y)
...
0 a
1 b
2 c
```

Reply:
Not quite sure I follow. I split this list:
`csq_values = picked_csqs[c][csq_idx].split("&")`
And for each value, find the index of that value in another list, `csq_severity_order`. So, unless enumerate can take take both lists, I think I am missing how this will work as it seems to take one one of them